src/simple_rules.ml

```diff
@@ -51,6 +51,12 @@ let copy_files sctx ~dir ~scope ~src_dir (def: Copy_files.t) =
     | Error (_pos, msg) ->
       Loc.fail (String_with_vars.loc def.glob) "invalid glob: %s" msg
   in
+  let file_tree = Super_context.file_tree sctx in
+  if not (File_tree.exists file_tree src_in_src) then
```

Comment:
What happens when `src_in_src` is a valid file? We might get a confusing error in that case. Perhaps it's worth adding a test case for this too.

Reply:
Indeed. I exposed and used `dir_exists` instead.